source/windows/mutex.c

```diff
@@ -0,0 +1,46 @@
+/*
+* Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License").
+* You may not use this file except in compliance with the License.
+* A copy of the License is located at
+*
+*  http://aws.amazon.com/apache2.0
+*
+* or in the "license" file accompanying this file. This file is distributed
+* on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+* express or implied. See the License for the specific language governing
+* permissions and limitations under the License.
+*/
+
+#include <aws/common/mutex.h>
+
+int aws_mutex_init(struct aws_mutex *mutex, struct aws_allocator *allocator) {
+    mutex->allocator = allocator;
+    InitializeCriticalSection(&mutex->mutex_handle);
```

Comment:
There's no error code for InitializeCriticalSection

https://msdn.microsoft.com/en-us/library/windows/desktop/ms683472(v=vs.85).aspx

I'll look into SRW locks

Reply:
"An SRW lock is the size of a pointer. The advantage is that it is fast to update the lock state. The disadvantage is that very little state information can be stored, so SRW locks cannot be acquired recursively"

Boom! sold

Reply:
Ah - critical section OOMs result in a structured exception thrown: https://blogs.msdn.microsoft.com/oldnewthing/20140911-00/?p=44103

Reply:
I'm moving to SRW regardless because a.) we don't have to support XP or server 2003 (and windows is a different scenario than supporting legacy linux because it's a commercial OS that reaches end of life), and b.) it's not reentrant which makes testing a hell of a lot easier.

Reply:
However, what baffles the hell out of me is that .NET 3.5 supports SRW and .NET 3.5 is deployed to windows xp where the api isn't available. My guess is that nt.dll exports it, and since dotnet is doing a dynamic bind, it picks it up, but it wasn't added to the headers until the windows sdk for vista.